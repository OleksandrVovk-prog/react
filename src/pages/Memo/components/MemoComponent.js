import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import styles from '../sass/MemoComponent.module.scss';

function MemoComponent({
  title, production, poster
}) {
  const { t } = useTranslation();
  return (
    <div className={styles.info}>
      <p className={styles.appIntro}>
        {t('title')}
        &nbsp;
        {title}
      </p>
      <p className={styles.appIntro}>
        {t('production')}
        &nbsp;
        {production}
      </p>
      <img
        className={styles.appBanner}
        src={poster}
        alt={title}
      />
    </div>
  );
}

export default React.memo(
  MemoComponent
);

MemoComponent.propTypes = {
  title: PropTypes.string.isRequired,
  production: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
};
