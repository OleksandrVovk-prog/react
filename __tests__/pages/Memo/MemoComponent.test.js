import React from 'react';
import { render } from '../../settings/test-utils';
import MemoComponent from '../../../src/pages/Memo/components/MemoComponent';

describe('<MemoComponent />', () => {
  let wrapper;
  const mockString = 'mock string';

  beforeEach(() => {
    wrapper = render(
      <MemoComponent
        production={mockString}
        title={mockString}
        poster={mockString}
      />
    );
  });

  test('should render MemoComponent component', () => {
    const { asFragment } = wrapper;
    expect(asFragment(wrapper)).toMatchSnapshot();
  });

  test('should render image', () => {
    const { getByRole } = wrapper;
    const Image = getByRole('img');
    expect(Image).toBeInTheDocument();
  });
});
