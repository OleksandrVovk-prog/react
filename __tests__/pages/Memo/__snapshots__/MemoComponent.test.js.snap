// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoComponent /> should render MemoComponent component 1`] = `
<DocumentFragment>
  <div
    class="info"
  >
    <p
      class="appIntro"
    >
      Title: mock string
    </p>
    <p
      class="appIntro"
    >
      Production: mock string
    </p>
    <img
      alt="mock string"
      class="appBanner"
      src="mock string"
    />
  </div>
</DocumentFragment>
`;
